posthog/api/test/test_decide.py

```diff
@@ -146,6 +146,82 @@ def test_feature_flags(self):
+        FeatureFlag.objects.create(
+            team=self.team, rollout_percentage=50, name="Beta feature", key="beta-feature", created_by=self.user,
+        )
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={"groups": [{"properties": [], "rollout_percentage": None}]},
+            name="This is a feature flag with default params, no filters.",
+            key="default-flag",
+            created_by=self.user,
+        )  # Should be enabled for everyone
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={
+                "groups": [{"properties": [], "rollout_percentage": None}],
+                "multivariate": {
+                    "variants": [
+                        {"key": "first-variant", "name": "First Variant", "rollout_percentage": 50},
+                        {"key": "second-variant", "name": "Second Variant", "rollout_percentage": 25},
+                        {"key": "third-variant", "name": "Third Variant", "rollout_percentage": 25},
+                    ],
+                    "fallback_variant_key": "third-variant",
+                },
+            },
+            name="This is a feature flag with multiple variants.",
+            key="multivariate-flag",
+            created_by=self.user,
+        )
+
+        with self.assertNumQueries(2):
+            response = self._post_decide(api_version=1)  # v1 functionality should not break
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertIn("beta-feature", response.json()["featureFlags"])
+            self.assertIn("default-flag", response.json()["featureFlags"])
+            self.assertIn("multivariate-flag", response.json()["featureFlags"])
+
+        with self.assertNumQueries(2):
+            response = self._post_decide(api_version=2)
+            self.assertTrue(response.json()["featureFlags"]["beta-feature"])
+            self.assertTrue(response.json()["featureFlags"]["default-flag"])
+            self.assertEqual(
+                "second-variant", response.json()["featureFlags"]["multivariate-flag"]
+            )  # randomly assigned by distinct_id hash
+
+    def test_feature_flags_v2_fallback(self):
+        self.team.app_urls = ["https://example.com"]
+        self.team.save()
+        self.client.logout()
+        Person.objects.create(team=self.team, distinct_ids=["example_id"], properties={"email": "tim@posthog.com"})
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={
+                "groups": [{"properties": [], "rollout_percentage": None}],
+                "multivariate": {
+                    "variants": [
+                        {"key": "first-variant", "name": "First Variant", "rollout_percentage": 10},
+                        {"key": "second-variant", "name": "Second Variant", "rollout_percentage": 0},
```

Comment:
From a UX perspective - what are your thoughts on the benefits of supporting a non-100% allocation w/ a fallback variant as opposed to ensuring multivariate flag variants sum to 100 @ creation time?

Reply:
I'm thinking if we enforce that allocations need to be at 100% then we don't need the concept of a fallback? But let me know if I'm missing something

Reply:
See [this comment](https://github.com/PostHog/posthog/pull/5626#discussion_r692498317) above — I like the concept of a fallback variant if we later decide to do more complex bucketing, like what is supported by |_aunch |)arkly.